SmartDeviceLink/SDLSoftButtonManager.m

```diff
@@ -190,36 +189,13 @@ - (void)setCurrentMainField1:(nullable NSString *)currentMainField1 {
 
 #pragma mark - RPC Responses
 
-- (void)sdl_registerResponse:(SDLRPCResponseNotification *)notification {
-    SDLRegisterAppInterfaceResponse *response = (SDLRegisterAppInterfaceResponse *)notification.response;
-
-    if (!response.success.boolValue) { return; }
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated"
-    if (response.displayCapabilities == nil) {
-        SDLLogE(@"RegisterAppInterface succeeded but didn't send a display capabilities. A lot of things will probably break.");
-        return;
-    }
-
-    self.softButtonCapabilities = response.softButtonCapabilities ? response.softButtonCapabilities.firstObject : nil;
-    self.displayCapabilities = response.displayCapabilities;
-#pragma clang diagnostic pop
-}
-
-- (void)sdl_displayLayoutResponse:(SDLRPCResponseNotification *)notification {
-    SDLSetDisplayLayoutResponse *response = (SDLSetDisplayLayoutResponse *)notification.response;
-    if (!response.success.boolValue) { return; }
-    if (response.displayCapabilities == nil) {
-        SDLLogE(@"SetDisplayLayout succeeded but didn't send a display capabilities. A lot of things will probably break.");
-        return;
-    }
-
-    self.softButtonCapabilities = response.softButtonCapabilities ? response.softButtonCapabilities.firstObject : nil;
-    self.displayCapabilities = response.displayCapabilities;
-
+- (void)sdl_displayCapabilityUpdate:(SDLSystemCapability *)systemCapability {
+    // we won't use the object in the parameter but the convenience method of the system capability manager
+    self.defaultMainWindowCapability = _systemCapabilityManager.defaultMainWindowCapability;
```

Comment:
I don't think there's a reason to use the getter from the `systemCapabilityManager`, the `systemCapability` that's passed should be equivalent because the handler here is only called after the merge finishes.

Reply:
fixed in 89d623d801bdbf805104544dc3f107b0b24f1675
though funny it looks